### ISSUE TITLE
Verification of extension function calls fails

### ISSUE THREAD

**Issue by raulbocanegra** [mockk/mockk#372]
### Prerequisites

Please answer the following questions for yourself before submitting an issue.

- [X] I am running the latest version
- [X] I checked the documentation and found no answer
- [X] I checked to make sure that this issue has not already been filed

### Expected Behavior

When an extension function has been called, a call to verify it should work instead of fail.

### Current Behavior

What is the current behavior?

When I call verify on an extension function that I am sure it is being called, tests fails.

#### Steps to Reproduce

Execute the unit test in the code below.

#### Context

* MockK version: 1.9.3
* OS: Ubuntu 18.04
* Kotlin version: 1.3.50
* JDK version: 1.8.0_222
* JUnit version: 4.12
* Type of test: unit test 

#### Failure Logs
```
Verification failed: call 1 of 1: Companion(#2).isEven(eq(0))) was not called
java.lang.AssertionError: Verification failed: call 1 of 1: Companion(#2).isEven(eq(0))) was not called
	at io.mockk.impl.recording.states.VerifyingState.failIfNotPassed(VerifyingState.kt:66)
	at io.mockk.impl.recording.states.VerifyingState.recordingDone(VerifyingState.kt:42)
	at io.mockk.impl.recording.CommonCallRecorder.done(CommonCallRecorder.kt:47)
	at io.mockk.impl.eval.RecordedBlockEvaluator.record(RecordedBlockEvaluator.kt:60)
	at io.mockk.impl.eval.VerifyBlockEvaluator.verify(VerifyBlockEvaluator.kt:30)
	at io.mockk.MockKDsl.internalVerify(API.kt:118)
	at io.mockk.MockKKt.verify(MockK.kt:139)
	at io.mockk.MockKKt.verify$default(MockK.kt:136)
	at MyNumberTest.verifyExtensionFunctionsAreCalled(MyNumberTest.kt:19)
	<46 internal calls>
	at java.lang.Thread.run(Thread.java:748)
```

#### Minimal reproducible code (the gist of this issue)

```kotlin
// -----------------------[ GRADLE DEFINITIONS ] -----------------------
val kotlinVersion = "1.3.50"
dependencies {
    implementation(kotlin("stdlib-jdk8"))
    testCompile("junit", "junit", "4.12")
    testCompile("org.jetbrains.kotlin:kotlin-test:$kotlinVersion")
    testCompile("org.jetbrains.kotlin:kotlin-test-junit:$kotlinVersion")
    testImplementation("io.mockk:mockk:1.9.3")
}
// -----------------------[ MY CODE STARTS HERE ] -----------------------
class MyNumber(private val value: Int) {
    fun isEven() : Boolean = value.isEven()
    companion object { 
        fun Int.isEven() : Boolean = this % 2 == 0
    }
}
// -----------------------[ MY TEST CODE STARTS HERE ] -----------------------
import io.mockk.every
import io.mockk.mockk
import io.mockk.verify
import org.junit.Assert.assertEquals
import org.junit.Test

class MyNumberTest {

    private val integerMock = mockk<Int>(relaxed = true)
    private val myNumber = MyNumber(integerMock)

    @Test
    fun verifyExtensionFunctionsAreCalled() {
        with(mockk<MyNumber.Companion>()) {
            every { integerMock.isEven() } returns true
            val even = myNumber.isEven()
            assertEquals(true, even)
            verify { integerMock.isEven() } // This line fails, but I think it should not because the assert above pass which means that the extension function has been called.
        }
    }
}

```

**Comment by oleksiyp** [comment]
Quite convoluted case... hard to understand what is wrong. 

First thing is that `MyNumber.Companion` is object and best to use `mockkObject` for it.

Second is that `Int` mocks shouldn't be possible to create at all.

**Comment by raulbocanegra** [comment]
Hi @oleksiyp If I remove the companion object, the `verify` call still fails. I used `Int` just for the example, you can use your own classes. The point is the `verify` calls does not work on extensions functions
```
class MyNumber(private val value: Int) {
    fun isEven() : Boolean = value.isEven()
    fun Int.isEven() : Boolean = this % 2 == 0    
}
```

**Comment by oleksiyp** [comment]
Member extension functions are not different than one that takes receiver as the first arg: `fun isEven($receiver: Int)`. From this point, you should think about how to mock it.

**Comment by raulbocanegra** [comment]
I can mock the extension function and it is called as expected, but the `verify` statement says it never was called when it really was.

**Comment by oleksiyp** [comment]
Anyway hard to understand for me this problem. `mockk<Int>` definitely will fail one or another way. And can't imagine the case for general class. A cleaner code can help.

**Comment by raulbocanegra** [comment]
Sorry for the example if it is that hard to understand. Let me rephrase it: When you inject a dependency of a class using the constructor and this dependency has an extension function, then when you use a mock for this dependency and mock the extension function, the extension function can't be verified. Calling verify to verify that the extension function was called always returns false. That's the issue I was trying to explain. The verify call should return true because the extension function has been called.
Thanks!
